tests/integration/startup.test.js

```diff
@@ -64,6 +64,43 @@ describe('daemon startup', function () {
             });
         });
     });
+
+    context('esr location', function () {
+
+        context('default', function () {
+
+            beforeEach('edit config file', function () {
+                editConfigFile(DAEMON_OBJ, null, ['swarm_info_esr_url']);
+            });
+
+            it('should default to infura', function (done) {
+                if (launchDaemon(['-c', DAEMON_OBJ.config_name]).stdout.includes('Connecting to: ropsten.infura.io...')) {
+                    done();
+                }
+            });
+
+        });
+
+        context('should be configurable with swarm_info_esr_url', function () {
+
+            beforeEach('edit config file', function () {
+                editConfigFile(DAEMON_OBJ, [['swarm_info_esr_url', 'http://127.0.0.1:8545']]);
+            });
```

Comment:
I would moved the daemon launch up to here, and combine assertions from the two below cases.

1. We spend less time for the test, since if we get one "Connecting" we don't have to wait as long (or at all) to assert both expectations.
2. We test technically more accurately by ensuring the same daemon process fully passes both expectations in one session.

Reply:
Moved. Also had some issues with execa, so just used spawn instead